darshan-util/pydarshan/tests/test_data_access_by_filesystem.py

```diff
@@ -0,0 +1,280 @@
+                                                                                                 verbose=verbose)
+        assert_series_equal(actual_read_groups, expected_read_groups)
+        assert_series_equal(actual_write_groups, expected_write_groups)
+    else:
+        with pytest.raises(NotImplementedError):
+            data_access_by_filesystem.unique_fs_rw_counter(report=report,
+                                                           filesystem_roots=filesystem_roots,
+                                                           file_id_dict=file_id_dict,
+                                                           processing_func=processing_func,
+                                                           mod=mod,
+                                                           verbose=verbose)
+
+
+@pytest.mark.parametrize("""file_rd_series,
+                            file_wr_series,
+                            bytes_rd_series,
+                            bytes_wr_series,
+                            filesystem_roots
+                         """, [
+
+     (pd.Series([3.0], index=pd.Index(['/p'], name='filesystem_root'), name='filepath'),
+      pd.Series([14.0], index=pd.Index(['/p'], name='filesystem_root'), name='filepath'),
+      pd.Series([2.145206e+09], index=pd.Index(['/p'], name='filesystem_root'), name='POSIX_BYTES_READ'),
+      pd.Series([1.010878e+12], index=pd.Index(['/p'], name='filesystem_root'), name='POSIX_BYTES_WRITTEN'),
+      ['/p'],
+         ),
+       ])
+def test_plot_data(file_rd_series, file_wr_series, bytes_rd_series, bytes_wr_series, filesystem_roots):
+    # test a few basic properties of the main plotting function
+    fig = plt.figure()
+    data_access_by_filesystem.plot_data(fig=fig,
+                                        file_rd_series=file_rd_series,
+                                        file_wr_series=file_wr_series,
+                                        bytes_rd_series=bytes_rd_series,
+                                        bytes_wr_series=bytes_wr_series,
+                                        filesystem_roots=filesystem_roots)
+    axes = fig.gca()
+    children = axes.get_children()
+    actual_list_text_in_fig = []
+
+    # accumulate text added via ax.text()
+    # by the function
+    for child in children:
+        if isinstance(child, matplotlib.text.Text):
+            actual_list_text_in_fig.append(child.get_text())
+
+    for expected_text_entry in [matplotlib.text.Text(0, 1, '# files read'),
+                                matplotlib.text.Text(0, 0, '# files written')]:
+        assert expected_text_entry.get_text() in actual_list_text_in_fig
+
+    # enforce invisibile right-side spine so that
+    # there is no overlap between value labels and
+    # the plot frame on the right side
+    for ax in fig.axes:
+        spines = ax.spines
+        right_spine_visibility = spines['right'].get_visible()
+        assert not right_spine_visibility
```

Comment:
Here's an example of regression testing a plotting issue that a reviewer brought up--this failed before I pushed in the "fix" to remove the right side spine. Not sure if the team will go with this approach, but it is one way to avoid value label overlap with the spine. Before/after visual below:

![sample-goodost darshan png_data_access_by_category_original](https://user-images.githubusercontent.com/7903078/124674604-4cbf9d80-de78-11eb-8564-31237ba5a9b1.png)
![sample-goodost darshan png_data_access_by_category](https://user-images.githubusercontent.com/7903078/124674610-4d583400-de78-11eb-983e-5ea3731e4aaf.png)

Of course, there's also no reason to have floats for the integer counter values, but one thing at a time...

Reply:
Well, we may want scientific notation on both sides in the end if there are cases with millions or billions of files, and not just large numbers of bytes read/written.